### PR TITLE
Organ removal ends metabolizing reagents in the organ

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request
If an organ contains reagents it will end metabolizing them on removal.

## Why It's Good For The Game
No more perma buffs for removal of an organ.

## Changelog
:cl:
fix: Organ removal stops reagent reactions on the host
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
called it

File: code/modules/surgery/organs/organ_internal.dm [hunk 1]
```diff
@@ -60,6 +60,13 @@
 
 //Special is for instant replacement like autosurgeons
 /obj/item/organ/proc/Remove(mob/living/carbon/M, special = FALSE)
+	//Stop any reagent metabolizing on organ destruction
+	if(reagents && iscarbon(owner))
+		var/mob/living/carbon/body = owner
+		for(var/bit in reagents.reagent_list)
+			var/datum/reagent/_reagent = bit
+			_reagent.on_mob_end_metabolize(body)
```
Comment (ATH1909) [inline code/modules/surgery/organs/organ_internal.dm:68]: and what if that reagent already wasn't being metabolized, due to the absence of a liver or the like? can I use this to repeatedly gain on_mob_end_metabolize() effects?

Comment (nightred) [reply code/modules/surgery/organs/organ_internal.dm:71]: yep you are not wrong, will get this fixed up.

File: code/modules/surgery/organs/organ_internal.dm [hunk 2]
```diff
@@ -60,6 +60,17 @@
 
 //Special is for instant replacement like autosurgeons
 /obj/item/organ/proc/Remove(mob/living/carbon/M, special = FALSE)
+	//Stop any reagent metabolizing on organ destruction
+	if(reagents && iscarbon(owner))
+		var/mob/living/carbon/body = owner
+		for(var/chem in reagents.reagent_list)
+			var/datum/reagent/_reagent = chem
```
Comment (Jared-Fogle) [inline code/modules/surgery/organs/organ_internal.dm:67]: This should probably just be named `reagent`. I personally do underscore names when I'm only using something for a typecast or its unused.

**Review by Mothblocks — DISMISSED** [review]
Test looks good!

File: code/modules/surgery/organs/organ_internal.dm (hunk 2 above)
Comment (nightred) [reply code/modules/surgery/organs/organ_internal.dm:67]: ha and i was like that is a good way to do names in type casts

File: code/modules/surgery/organs/organ_internal.dm [hunk 3]
```diff
@@ -60,6 +60,17 @@
 
 //Special is for instant replacement like autosurgeons
 /obj/item/organ/proc/Remove(mob/living/carbon/M, special = FALSE)
+	//Stop any reagent metabolizing on organ destruction
+	if(reagents && iscarbon(owner))
+		var/mob/living/carbon/body = owner
+		for(var/chem in reagents.reagent_list)
+			var/datum/reagent/reagent = chem
+			if(!reagent.metabolizing)
```
Comment (ATH1909) [inline code/modules/surgery/organs/organ_internal.dm:68]: the on_mob_delete() call needs to be moved to be BEFORE this check

Comment (nightred) [reply code/modules/surgery/organs/organ_internal.dm]: Shit thought i was clever with that early continue.

**Review by Timberpoes — APPROVED** [review]
From a brief look over the code, seems to do what it says on the tin. Jared seemed to be happy with the unit tests. Approving this.

**Pull request merged**